process.md

```diff
@@ -0,0 +1,338 @@
+# Console Processes
+
+- [Introduction](#introduction)
+- [Running Processes](#running-processes)
+    - [Command](#command)
+    - [Path](#path)
+    - [Timeout](#timeout)
+    - [Error Handling](#error-handling)
+- [Concurrent Processes](#concurrent-processes)
+- [Asynchronous Processes](#asynchronous-processes)
+- [Macros](#macros)
+- [Testing](#testing)
+    - [Faking Process Results](#faking-results)
+    - [Inspecting Process Requests](#inspecting-results)
+
+<a name="introduction"></a>
+## Introduction
+
+When building web applications, it's common to have the need of running operating system tasks via command-line processes. Thankfully, Laravel provides an expressive, minimal API around Symfony's Process component, allowing you to quickly run processes in your Laravel application.
+
+Laravel's wrapper around Symfony's Process component is focused on its most common use cases, testing, and a wonderful developer experience.
+
+<a name="running-processes"></a>
+## Running Processes
+
+To run a process, you may use the `run` method provided by the `Process` facade. Let's examine how to run a basic `ls` command:
+
+    use Illuminate\Support\Facades\Process;
+
+    $result = Process::run('ls');
+
+    $result->ok(); // true
+    $result->output(); // my-file-1, my-file-2
+    $result->toArray(); // ["my-file-1", "my-file-2"]
+
+The `run` method returns an instance of `Illuminate\Console\Contracts\ProcessResult`, which provides a variety of methods that may be used to inspect the process result:
+
+    $result = Process::run('ls');
+
+    $result->output(): string;
+    $result->toArray(): array;
+    $result->errorOutput(): string;
+    $result->exitCode(): int;
+    $result->ok(): bool;
+    $result->failed(): bool;
+
+The `Illuminate\Console\Contracts\ProcessResult` object also implements the PHP `ArrayAccess` interface, allowing you to iterate over each line of the output. The output, is "exploded" via the `\n` character:
+
+    $files = Process::run('ls');
+
+    foreach ($files as $file) {
+        // ...
+    }
+
+    return $files[0]; // my-file-1
+
+<a name="accessing-the-output-at-real-time"></a>
+#### Accessing the output at real-time
+
+By default, when using the `output` method, Laravel waits for the process to be finished, so you can have access to the entire process's output:
+
+    $result = Process::run('echo 1; sleep 1; echo 2');
+
+    return $result->output(); // 1, 2
+
+If you wish to access the process's output at real-time, you may use a closure as the second argument of the `run` method:
+
+    Process::run('echo 1; sleep 1; echo 2', function ($output) {
+        dump($output); // Dumps "1", and after a second dumps "2"
+    });
+
+    return $result->output(); // 1, 2
+
+In addition, if you with to know if the given output is from the type `stderr` or `stderr`, you may use the second argument of given closure:
```

Comment:
I think here you meant `if you want to know` instead of `if you with to know` ?

Reply:
Fixed.